internal/cli/internal/command/machines/flaps.go

```diff
@@ -0,0 +1,150 @@
+package machines
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/superfly/flyctl/api"
+	"github.com/superfly/flyctl/flyctl"
+	"github.com/superfly/flyctl/internal/client"
+	"github.com/superfly/flyctl/pkg/agent"
+)
+
+type Machine struct {
+	*api.Machine
+}
+
+type BuildConfigInput struct {
+	AppID   string            `json:"appId,omitempty"`
+	OrgSlug string            `json:"organizationId,omitempty"`
+	Region  string            `json:"region,omitempty"`
+	Config  api.MachineConfig `json:"config"`
+}
+
+type FlapsClient struct {
+	ctx       context.Context
+	app       *api.App
+	peerIP    string
+	authToken string
+}
+
+func NewFlapsClient(ctx context.Context, app *api.App) (*FlapsClient, error) {
+	client := client.FromContext(ctx).API()
+	agentclient, err := agent.Establish(ctx, client)
+	if err != nil {
+		return nil, fmt.Errorf("error establishing agent: %w", err)
+	}
+
+	dialer, err := agentclient.Dialer(ctx, app.Organization.Slug)
+	if err != nil {
+		return nil, fmt.Errorf("ssh: can't build tunnel for %s: %s", app.Organization.Slug, err)
+	}
+
+	return &FlapsClient{
+		ctx:       ctx,
+		app:       app,
+		peerIP:    resolvePeerIP(dialer.State().Peer.Peerip),
+		authToken: flyctl.GetAPIToken(),
+	}, nil
+}
+
+func (f *FlapsClient) Launch(builder api.LaunchMachineInput) (*Machine, error) {
```

Comment:
Should we be passing the context down to these methods, and to the client? For cancelability, and also to use iostreams instead of printing directly to stdout.

Reply:
Also, might we derive value from the flaps package existing on its own in `pkg` instead of inside the command structure?